### PR TITLE
feat: add signup-signin functionality

### PR DESCRIPTION
**Pull request by PuneetP16**
## To Check on Live Preview: 
https://deploy-preview-6--magnificent-store-r.netlify.app/
Login, Logout and SignUp functionality.
Loader Functionality on Login.
On SignUp, navigating to login component and populating field with email and password.
when remember me checkbox is checked, saving the login details to localStorage and setting it as initial login details.

## Files to Check
Login.jsx
SignUp.jsx
Loader.jsx
userContext.jsx
authContext.jsx
signin.js
signup.js
app.js

### CODE REVIEW

**Review by saurabhchirde — APPROVED** [review]
Couldn't find any issue, Looks good to merge.

File: src/App.js [hunk 1]
```diff
@@ -11,19 +11,29 @@ import {
 	ProductListing,
 	NotFound,
 } from "./pages";
-import { Footer, Header } from "./components";
+import { Footer, Header, Loader } from "./components";
+import { useAuth } from "./contexts";
 
 function App() {
 	const currentPath = useLocation().pathname;
```
Comment (Debarshi95) [inline src/App.js]: Suggestion, you can restructure **pathname** here. If it's undefined initially, you can add default value to it. 
Ex:
```
const { pathname =' '}=useLocation();
```

**Review by Debarshi95 — APPROVED** [review]
Made a comment. Some suggestions, use a lint guide like eslint, and formatter such as prettier to format your code. A considerable amount of optimizations related to re-rendering is required in a few components, but for now you can merge the PR.

File: src/App.js (hunk 1 above)
Comment (PuneetP16) [reply src/App.js]: Updated. Thanks for pointing out.

File: src/components/UI/Loader/Loader.css [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	padding: 1rem;
+}
+
+.modal_container {
+	position: absolute;
+	top: 0;
+	bottom: 0;
+	right: 0;
+	left: 0;
+	z-index: 999;
```
Comment (tarunsankhla) [inline src/components/UI/Loader/Loader.css:24]: You can create a variable for zindex stack

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 3]
```diff
@@ -1,18 +1,56 @@
-import { Link } from "react-router-dom";
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
```
Comment (tarunsankhla) [inline src/pages/Authentication/SignUp/SignUp.jsx:2]: Can have better components name inputtype one two three is not much easily understandable

**Review by tarunsankhla — APPROVED** [review]
Over all the code is great and logic is also good

File: src/components/UI/Loader/Loader.css (hunk 2 above)
Comment (PuneetP16) [reply src/components/UI/Loader/Loader.css:24]: ok, will do that.

File: src/pages/Authentication/SignUp/SignUp.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/pages/Authentication/SignUp/SignUp.jsx:2]: Will look into it.

**Pull request merged**